.github/workflows/publish-images.yaml

```diff
@@ -0,0 +1,48 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    tags:
+      - "*"
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        image: [dask-gateway, dask-gateway-server]
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
```

Comment:
Can a comment be made to clarify what this does? Are you looking at the latest published image, and applies the same tags/labels to the new image?

Reply:
> GitHub Action to extract metadata from Git reference and GitHub events.

Ah, so this is refining Git and GitHub metadata, so we can use it to tag/label our built docker images.

I'm not sure what git labels are extracted, and if it would make sense to set them on the built image as well - but it makes sense to extract versions from the git tag I think.

Given how this is configured, it will result in `v1.2.3` and `latest` tag if a tag `v1.2.3` is pushed. It can also be configured to emit tags `v1.2.3`, `v1.2`, and `latest` for example.

The use of this action, is the use of what the https://github.com/jupyterhub/action-major-minor-tag-calculator#readme does it seems.

Reply:
I needed to read up a lot about this action. Let's try it out!